src/main/java/com/joyent/manta/monitor/MantaMonitorJerseyServer.java

```diff
@@ -141,10 +220,37 @@ private void configureServer() {
             httpsConfig.addCustomizer(src);
 
             SslContextFactory sslContextFactory = new SslContextFactory();
-            sslContextFactory.setKeyStorePath(System.getenv("KEYSTORE_PATH"));
+
+            String userHomePath = System.getProperty("user.home");
+
+            URI keystoreURI = getURIFromString(System.getenv("KEYSTORE_PATH"));
+
+            if ("manta".equals(keystoreURI.getScheme())) {
+                // Make the keystore file available for the sslContextFactory, by
+                // reading the remote file and writing it to a file on the host file system.
+                File keystoreTargetFile = new File(userHomePath + File.separator
+                        + MANTA_SUBDIR + File.separator + "keystore");
+                writeInputStreamToTarget(keystoreURI, keystoreTargetFile);
+                sslContextFactory.setKeyStorePath(keystoreTargetFile.getPath());
+            } else {
+                // Read the keystore directly from the local file system
+                sslContextFactory.setKeyStorePath(System.getenv("KEYSTORE_PATH"));
+            }
+
+            URI trustStoreURI = getURIFromString(System.getenv("TRUSTSTORE_PATH"));
+            if ("manta".equals(trustStoreURI.getScheme())) {
+                // Make the truststore file available for the sslContextFactory, by
+                // reading the remote file and writing it to a file on the host file system.
+                File trustStoreTargetFile = new File(userHomePath + File.separator
+                        + MANTA_SUBDIR + File.separator + "truststore");
```

Comment:
Let's put all of the config files by default under `$HOME/.manta-monitor/`. I think that is the last change before merge.

Reply:
You mean even the CONFIG_FILE? It is right now read by the ```MantaObjectMapper``` and not required to be stored locally. The only files required locally are the KEYSTORE and the TRUSTSTORE files which are being saved under /.manta-monitor. Actually, since these are TLS related files, the sub directory can be named likewise, for eg: /.manta-monitor-tls?

Reply:
My main concern is that our default path is right in the middle of someone's home directory. This is a really bad practice and we shouldn't be doing it.

Let's put all files default paths under `~/.manta-monitor`. If we want to have another classification, we can create `~/.manta-monitor/tls`.

Reply:
How about putting the files under /tmp. like /tmp/.manta-monitor/? This avoids resolving any relative paths.

Reply:
Is it ok if the files are deletes when the system restarts or reboots? If so, then `/tmp` is fine.

Reply:
I think better alternative would be ```System.getProperty("user.dir")/.manta-monitor```. This way it will write to ```/opt/manta-monitor``` which is the default working directory of manta-monitor.

Reply:
That makes more sense. Be sure to document the defaults.

Reply:
Yes. Thanks!